src/coreclr/src/pal/src/configure.cmake

```diff
@@ -824,23 +838,25 @@ check_cxx_source_runs("
 int main(void) {
   double pi = 3.14159265358979323846;
   double result;
+  volatile double y = 0.0;
+  volatile double x = 0.0;
 
-  result = atan2(0.0, -0.0);
+  result = atan2(y, -x);
   if (fabs(pi - result) > 0.0000001) {
```

Comment:
This (and others checking against a "close enough" result) should ideally get cleaned up as well. IEEE floating-point is deterministic, if nothing else, and so the result should be exactly `Math.PI` (which is the nearest representation of "true pi").
I'll log a bug and see about following up after this gets merged.

Reply:
Logged https://github.com/dotnet/runtime/issues/35341